cli/src/lib.rs

```diff
@@ -636,31 +562,28 @@ mod tests {
 
     mod config_integration {
         use assertables::{assert_contains, assert_contains_as_result};
+        use iroha_config::parameters::user_layer::RootPartial as PartialUserConfig;
         use iroha_crypto::KeyPair;
         use iroha_genesis::{ExecutorMode, ExecutorPath};
         use iroha_primitives::addr::socket_addr;
         use path_absolutize::Absolutize as _;
 
         use super::*;
 
-        fn config_factory() -> Result<ConfigurationProxy> {
-            let key_pair = KeyPair::generate()?;
+        fn config_factory() -> Result<PartialUserConfig> {
+            let (pubkey, privkey) = KeyPair::generate()?.into();
 
-            let mut base = ConfigurationProxy {
-                chain_id: Some(ChainId::new("0")),
+            let mut base = PartialUserConfig::default();
 
-                public_key: Some(key_pair.public_key().clone()),
-                private_key: Some(key_pair.private_key().clone()),
+            base.iroha.chain_id.set(ChainId::from("0"));
```

Comment:
```suggestion
            base.iroha.set_chain_id(ChainId::from("0"));
```

we should stick to conventions

Reply:
`{field}.set()` comes not from our conventions, but from keeping in mind that there will be a separate configuration library which wraps all fields in a `UserField<T>` struct. `.set()` comes from it.

Let's discuss `UserField<T>` where it is defined now, in `iroha_config_base`.

Reply:
the problem is that we're breaking our conventions from `data_model` where we have:
* getter = `struct.field()`
* setter = `struct.set_field(value)`

Additionally, this doesn't seem to play well with my comment [here](https://github.com/hyperledger/iroha/pull/4239#discussion_r1469744711)

This can be macro generated as is done with `getset`

Reply:
You say that it breaks conventions established by the `data_model`. I don't argue with that.

The point is that this comes from `iroha_config_base`, which I made having an idea in mind that it will be an external crate from Iroha. There is no need to follow conventions of Iroha there.

I feel it's not valid to generate here `set_field` for each partial's field when you can simply assign values directly. However, in these cases there are `UserField`s, which are simpler to re-set with `::set` method.. Not sure if I make sense here.

Reply:
> The point is that this comes from `iroha_config_base`, which I made having an idea in mind that it will be an external crate from Iroha. There is no need to follow conventions of Iroha there.
>
> I feel it's not valid to generate here `set_field` for each partial's field when you can simply assign values directly. However, in these cases there are `UserField`s, which are simpler to re-set with `::set` method.. Not sure if I make sense here.

we need to cross reference the discussion from [here](https://github.com/hyperledger/iroha/pull/4239#discussion_r1486036549). I listed the reasons for hiding internal of the config structs there. It's not about following Iroha conventions, it's about establishing a robust set of conventions that will work for any project. And making config structs immutable will prevent user errors

Reply:
create #4282